src/wallet/test/spend_tests.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <policy/fees.h>
+#include <validation.h>
+#include <wallet/coincontrol.h>
+#include <wallet/test/util.h>
+#include <wallet/test/wallet_test_fixture.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(spend_tests, WalletTestingSetup)
+
+BOOST_FIXTURE_TEST_CASE(SubtractFee, TestChain100Setup)
```

Comment:
No need to override fixture declared for test suite. You can do

```
BOOST_FIXTURE_TEST_SUITE(spend_tests, TestChain100Setup)
BOOST_AUTO_TEST_CASE(SubtractFee)
```

or

```
BOOST_AUTO_TEST_SUITE(spend_tests)
BOOST_FIXTURE_TEST_CASE(SubtractFee, TestChain100Setup)
```

Reply:
It's better because it avoids unnecessary information. `WalletTestingSetup` is never used in the test suite.

There is no point in setting TEST_SUITE fixture to `WalletTestingSetup` when the only test case in the test suite overrides the fixture and uses `TestChain100Setup`.  It's just unused information in the test that could potentially confuse readers.

Same goes for `wallet_tests`. Every test case there defines there own fixture explicitly and don't inherit it from the test suite. I have no idea why it's like that

Reply:
I understand the motivation now and yes I agree. WalletTestingSetup can be overkill because even if you are testing wallet code, you are likely testing smaller functions and classes that don't need the full setup. It's nicer in general to use simpler fixtures for efficiency and easier comprehension, and would be good to see use of fixtures cleaned up and simplified in future PRs. Consistency between spend_tests and wallet_tests here may at least help with that.